AWSS3Tests/AWSS3TransferUtilityTests.swift

```diff
@@ -2322,7 +2322,8 @@ class AWSS3TransferUtilityTests: XCTestCase {
                 
                 if let HTTPResponse = task.response {
                     XCTAssertEqual(HTTPResponse.statusCode, 200)
-                    
+                    // Sleep for 2 sec so that the url created will be accessible.
+                    sleep(2)
```

Comment:
Instead, make this a loop that does a `headObject` until the metadata comes back/doesn't throw an exception, then do the download.

Reply:
Can we leverage [expectations](https://developer.apple.com/documentation/xctest/asynchronous_tests_and_expectations/testing_asynchronous_operations_with_expectations) in this case? A `waitUntil` utility could fit this case well

Reply:
Thanks for the suggestion and I updated the test with a loop.

@drochetti Not sure whether we can use expectation for a pooling call. Maybe we might need to create our own implementation of expectation?